packages/runtime/src/components/page/Page.tsx

```diff
@@ -1,19 +1,19 @@
 'use client'
 
 import { BodySnippet } from './BodySnippet'
-import { DocumentReference } from '../../runtimes/react'
-import { createDocumentReference } from '../../state/react-page'
+import { DocumentRoot } from '../../runtimes/react'
+import { type Document } from '../../state/react-page'
 import { MakeswiftPageDocument } from '../../next'
 import { useRouterLocaleSync } from '../hooks/useRouterLocaleSync'
 import { usePageSnippets } from '../hooks/usePageSnippets'
 import { PageHead } from './PageHead'
 
 type Props = {
-  pageDocument: MakeswiftPageDocument
-  documentKey: string
+  page: MakeswiftPageDocument
+  rootDocument: Document
 }
 
-export function Page({ pageDocument: page, documentKey }: Props): JSX.Element {
+export function Page({ page, rootDocument }: Props): JSX.Element {
```

Comment:
Take the whole root document instead of the `documentKey` here so that we don't have to re-fetch it from the store (the usage site does have the document already).